libsolidity/formal/SolverInterface.h

```diff
@@ -36,6 +36,14 @@ namespace solidity
 namespace smt
 {
 
+enum class SMTSolverChoice
```

Comment:
To be more future proof, maybe:
```suggestion
struct SMTSolverChoice
{
  bool cvc4 = false;
  bool z3 = false;
  static constexpr SMTSolverChoice All { true, true };
  static constexpr SMTSolverChoice CVC4 { true, false };
  static constexpr SMTSolverChoice Z3 { false, true };
  static constexpr SMTSolverChoice None { false, false };
};
```
Then instead of checking ``choice == All || choice == Z3`` you can just check ``choice.z3``...

Reply:
Right

Reply:
```
struct SMTSolverChoice
{
  bool cvc4 = false;
  bool z3 = false;
  static constexpr SMTSolverChoice All() { return { true, true }; }
  static constexpr SMTSolverChoice CVC4() { return { true, false }; }
  static constexpr SMTSolverChoice Z3() { return { false, true }; }
  static constexpr SMTSolverChoice None() { return { false, false }; }
};
```